lib/elixir/test/elixir/protocol_test.exs

```diff
@@ -375,6 +381,10 @@ defmodule Protocol.ConsolidationTest do
     assert {{:ok, 1}, _, :def, [{:term, _, nil}], "Ok"} = List.keyfind(docs, {:ok, 1}, 0)
   end
 
+  test "consolidation keeps deprecated" do
+    assert [{{:ok, 1}, "Reason"}] = Sample.__info__(:deprecated)
```

Comment:
We need to check that it also keeps the deprecated chunk. My impression is that it does not, given the discussion on #7117.

Reply:
Sorry, I see my error now. I was using the `:code.get_object_code(Sample)` and it was returning the old binary I think. When I checked that inside of the tests for the docs the same thing happens in `Code.get_docs`, I thought it somehow the compiler was preserving the binary.